get_stories.php

```diff
@@ -90,35 +90,37 @@ public static function nprstory_cron_pull() {
 	 * Function to convert an alleged NPR story URL or ID into a story ID, then request it
 	 */
 	public function load_page_hook() {
+		// if the current user shouldn't be doing this, fail
+		if ( !current_user_can( 'edit_posts' ) ) {
+			wp_die(
+				__( 'You do not have permission to edit posts, and therefore you do not have permission to pull posts from the NPR API' ),
+				__( 'NPR Story API Error' ),
+				403
+			);
+		}
+
 		// find the input that is allegedly a story id
 		// We validate these later
 		if ( isset( $_POST ) && isset( $_POST[ 'story_id' ] ) ) {
-			$story_id =  $_POST[ 'story_id' ] ;
-			if ( isset( $_POST['publishNow'] ) ) {
-				$publish = true;
-			}
-			if ( isset( $_POST['createDaft'] ) ) {
-				$publish = false;
-			}
 			if ( !check_admin_referer( 'nprstory_nonce_story_id', 'nprstory_nonce_story_id_field' ) ) {
 				wp_die(
 					__( 'Nonce did not verify in DS_NPR_API::load_page_hook. Are you sure you should be doing this?' ),
 					__( 'NPR Story API Error' ),
 					403
 				);
 			}
+			$story_id = sanitize_text_field( $_POST[ 'story_id' ] );
+			if ( isset( $_POST['publishNow'] ) ) {
+				$publish = true;
+			}
+			if ( isset( $_POST['createDaft'] ) ) {
```

Comment:
is 'createDaft' a typo? @jwcounts

Reply:
It's a typo in the variable name, but yeah, it's been there a while and is consistent. I decided not to mess with it. The text on the visible button that's connected to is correct, though.